token/program-2022/src/processor.rs

```diff
@@ -1292,6 +1292,83 @@ impl Processor {
         Ok(())
     }
 
+    /// Withdraw Excess Lamports is used to recover Lamports transfered to any TokenProgram owned account
+    /// by system program's transfer instruction by moving them to WrappedSol ATA owned by the authority
+    /// of the source account.
+    pub fn process_withdraw_excess_lamports(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+
+        let destination_info = next_account_info(account_info_iter)?;
+        let source_info = next_account_info(account_info_iter)?;
+        let authority_info = next_account_info(account_info_iter)?;
+
+        let source_data = source_info.data.borrow();
```

Comment:
Since `validate_owner`  also borrows the data for `authority_info`, in the case of a multisig, you may get a double-borrow which would panic.

Could you add tests to successfully recover lamports from each a mint, multisig, and token account? You should also add tests for self-owned accounts, ie a token account or mint that is owned by itself. The multisig case should be covered, thankfully